test/suite/test_import01.py

```diff
@@ -0,0 +1,188 @@
+            "Tmplog" not in file_name and "Preplog" not in file_name:
+            shutil.copy(os.path.join(old_dir, file_name), new_dir)
+
+    def test_file_import(self):
+        original_db_file = 'original_db_file'
+        uri = 'file:' + original_db_file
+
+        create_config = 'allocation_size=512,key_format=u,log=(enabled=true),value_format=u'
+        self.session.create(uri, create_config)
+
+        key1 = b'1'
+        key2 = b'2'
+        key3 = b'3'
+        key4 = b'4'
+        value1 = b'\x01\x02aaa\x03\x04'
+        value2 = b'\x01\x02bbb\x03\x04'
+        value3 = b'\x01\x02ccc\x03\x04'
+        value4 = b'\x01\x02ddd\x03\x04'
+
+        # Add some data.
+        self.update(uri, key1, value1, 10)
+        self.update(uri, key2, value2, 20)
+
+        # Perform a checkpoint.
+        self.session.checkpoint()
+
+        # Add more data.
+        self.update(uri, key3, value3, 30)
+        self.update(uri, key4, value4, 40)
+
+        # Perform a checkpoint.
+        self.session.checkpoint()
+
+        # Export the metadata for the table.
+        c = self.session.open_cursor('metadata:', None, None)
+        original_db_file_config = c[uri]
+        c.close()
+
+        self.printVerbose(3, '\nFILE CONFIG\n' + original_db_file_config)
+
+        # Contruct the config string.
+        import_config = 'import=(enabled,repair=false,file_metadata=(' + \
+            original_db_file_config + '))'
+
+        if self.import_type == 'same':
+            # Try to import the file even though it already exists in our database.
+            # We should get an error back.
+            self.assertRaisesException(wiredtiger.WiredTigerError,
+                lambda: self.session.create(uri, import_config))
+            return
+
+        # Close the connection.
+        self.close_conn()
+
+        # Create a new database and connect to it.
+        newdir = 'IMPORT_DB'
+        shutil.rmtree(newdir, ignore_errors=True)
+        os.mkdir(newdir)
+        self.conn = self.setUpConnectionOpen(newdir)
+        self.session = self.setUpSessionOpen(self.conn)
+
+        # Simulate importing a file into an existing database.
+        # Make a bunch of files and fill them with data.
+        if self.import_type == 'existing':
+            self.populate()
+
+        # Copy over the datafiles for the object we want to import.
+        self.copy_file(original_db_file, '.', newdir)
+
+        # Import the file.
+        self.session.create(uri, import_config)
+
+        # Verify object.
+        self.session.verify(uri)
+
+        # Check that the previously inserted values survived the import.
+        self.check(uri, key1, value1, 10)
+        self.check(uri, key2, value2, 20)
+        self.check(uri, key3, value3, 30)
+        self.check(uri, key4, value4, 40)
+
+        # Compare checkpoint information.
+        # TO-DO: We really want to compare the complete metadata, but we need to split and
```

Comment:
We should figure out how to do this before merging. Let me look at it a bit today.

Reply:
5caf401